src/main/java/net/sf/jabref/logic/util/DOI.java

```diff
@@ -107,6 +107,23 @@ public DOI(String doi) {
     }
 
     /**
+     * Creates an Optional<DOI> from various schemes including URL, URN, and plain DOIs.
+     *
+     * Useful for suppressing the <c>IllegalArgumentException</c> of the Constructor
+     * and checking for Optional.isPresent() instead.
+     *
+     * @param doi the DOI string
+     * @return an Optional containing the DOI or an empty Optional
+     */
+    public static Optional<DOI> build(Optional<String> doi) {
```

Comment:
I kind of like it here since it leads to a more fluid API. See also the discussion at http://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments